ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/peers/EnodeURL.java

```diff
@@ -32,7 +32,7 @@
 import com.google.common.primitives.Ints;
 import org.apache.tuweni.bytes.Bytes;
 
-public class EnodeURL {
+public class EnodeURL implements org.hyperledger.besu.plugin.data.EnodeURL {
```

Comment:
should the `EnodeURL` class be named more about what implementation is doing? for p2p, etc

Reply:
This is a difficult one, the `EnodeURL` introduces a circular dependency between the plugin and the implementation. That's why I introduced the interface. From an OO naming point of view, nothing really makes sense. Hungarian`IEnodeURL` isn't used anywhere else in besu `Impl` is but would mean there would be about 600 changes!

Reply:
I think is worth changing (IntelliJ can handle the 600 changes, but if you prefer, can be done in a separate PR).

should the interface `EnodeURL` live inside plugin data? maybe needs to be something more common (not only for plugins).
moreover, in the class `EnodeURL`, is there anything specific about this implementation, like used for `p2p` or `peer`? instead of using `Impl`.

Reply:
> I think is worth changing (IntelliJ can handle the 600 changes, but if you prefer, can be done in a separate PR).
100% a separate or first to get rid of the noise on this commit.
> should the interface EnodeURL live inside plugin data? maybe needs to be something more common (not only for plugins).
Perhaps not, we would need to create a new module to contain such things. 
>  moreover, in the class EnodeURL, is there anything specific about this implementation, like used for p2p or peer? instead of using Impl.
Nope :disappointed: